arm-customer-insights/2016-01-01/swagger/customer-insights.json

```diff
@@ -0,0 +1,4735 @@
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/widgetTypes/{widgetTypeName}": {
+      "get": {
+        "tags": [
+          "WidgetTypes"
+        ],
+        "operationId": "WidgetTypes_Get",
+        "x-ms-examples": { 
+          "WidgetTypes_Get": { "$ref": "../examples/WidgetTypesGet.json" } 
+        },
+        "description": "Gets a widget type in the specified hub.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "widgetTypeName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the widget type."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/WidgetTypeResourceFormat"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/views": {
+      "get": {
+        "tags": [
+          "Views"
+        ],
+        "operationId": "Views_ListByHub",
+        "x-ms-examples": { 
+          "Views_ListByHub": { "$ref": "../examples/ViewsListByHub.json" } 
+        },
+        "description": "Gets all available views for given user in the specified hub.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "userId",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "The user ID. Use * to retreive hub level views."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "sc
```

Comment:
>"The purpose of the swagger spec is to correctly and completely represent underlying APIs"

So yes you must specify names of the model property as they'd appear on the wire. Whenever you decide to change the casing, it'll be a breaking change. 

Note: Keys in a json object are case-sensitive. `{id: 'fdfdfdf', ID: 'dfdfdfdfdf' }` are two different keys/properties in the json object. Hence, casing is very important during [de]serialization.